src/QsCompiler/CompilationManager/EditorSupport/SymbolInfo.cs

```diff
@@ -277,13 +342,168 @@ bool OverlapsWithPosition(Range symRange)
+                QsExpressionKind.ADD add => Binary(add.Item1, add.Item2),
+                QsExpressionKind.SUB sub => Binary(sub.Item1, sub.Item2),
+                QsExpressionKind.MUL mul => Binary(mul.Item1, mul.Item2),
+                QsExpressionKind.DIV div => Binary(div.Item1, div.Item2),
+                QsExpressionKind.MOD mod => Binary(mod.Item1, mod.Item2),
+                QsExpressionKind.POW pow => Binary(pow.Item1, pow.Item2),
+                QsExpressionKind.EQ eq => Binary(eq.Item1, eq.Item2),
+                QsExpressionKind.NEQ neq => Binary(neq.Item1, neq.Item2),
+                QsExpressionKind.LT lt => Binary(lt.Item1, lt.Item2),
+                QsExpressionKind.LTE lte => Binary(lte.Item1, lte.Item2),
+                QsExpressionKind.GT gt => Binary(gt.Item1, gt.Item2),
+                QsExpressionKind.GTE gte => Binary(gte.Item1, gte.Item2),
+                QsExpressionKind.AND and => Binary(and.Item1, and.Item2),
+                QsExpressionKind.OR or => Binary(or.Item1, or.Item2),
+                QsExpressionKind.BOR bOr => Binary(bOr.Item1, bOr.Item2),
+                QsExpressionKind.BAND bAnd => Binary(bAnd.Item1, bAnd.Item2),
+                QsExpressionKind.BXOR bXor => Binary(bXor.Item1, bXor.Item2),
+                QsExpressionKind.LSHIFT lShift => Binary(lShift.Item1, lShift.Item2),
+                QsExpressionKind.RSHIFT rShift => Binary(rShift.Item1, rShift.Item2),
+                QsExpressionKind.CONDITIONAL cond => Many(new[] { cond.Item1, cond.Item2, cond.Item3 }),
+                QsExpressionKind.CopyAndUpdate update => Binary(update.Item1, update.Item3),
+                QsExpressionKind.UnwrapApplication unwrap => DeclarationsInExpression(unwrap.Item, position),
+                QsExpressionKind.AdjointApplication adj => DeclarationsInExpression(adj.Item, position),
+                QsExpressionKind.ControlledApplication ctl => DeclarationsInExpression(ctl.Item, position),
+                QsExpressionKind.CallLikeExpression call => Binary(call.Item1, call.Item2),
+                QsExpressionKind.SizedArray sizedArray => Binary(sizedArray.value, sizedArray.size),
+                QsExpressionKind.Lambda lambda => Lambda(lambda.Item),
+                _ => Enumerable.Empty<LocalVariableDeclaration<string>>(),
+            };
+
+            IEnumerable<LocalVariableDeclaration<string>> Binary(TypedExpression e1, TypedExpression e2) =>
+                DeclarationsInExpression(e1, position).Concat(DeclarationsInExpression(e2, position));
+
+            IEnumerable<LocalVariableDeclaration<string>> Many(IEnumerable<TypedExpression> es) =>
+                es.SelectMany(e => DeclarationsInExpression(e, position));
+
+            IEnumerable<LocalVariableDeclaration<string>> Lambda(Lambda<TypedExpression> lambda)
+            {
+                var inputType = CallableInputType(expression.ResolvedType);
+                var inferred = new InferredExpressionInformation(false, true);
```

Comment:
I am curious about your thinking on always setting hasLocalQuantumDependency to true. It seems reasonable as a pessimistic assumption, given that hasLocalQuantumDependency always has to assume that there is a dependency when there could be. I think it merits a comment in the code calling this out.

Reply:
Added comment.